test/smoke/testApps/HeartBeat/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HeartBeatTest.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.applicationinsights.smoketest;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.MetricData;
+
+import org.junit.Test;
+
+public class HeartBeatTest extends AiSmokeTest {
+
+    @Test
+    @TargetUri(value = "/index.jsp")
+    public void testHeartBeat() throws Exception {
+        List<Envelope> metrics = mockedIngestion.waitForItems(getMetricPredicate("HeartbeatState"), 2, 70, TimeUnit.SECONDS);
+        assertEquals(2, metrics.size());
```

Comment:
We should assert on some of the field values. This just tests that the config works.

@dhaval24, what should be checked?

Reply:
We can assert on basic fields are added into the custom dimensions of HeartBeat payload should have values for : jreVersion, sdkVersion, osVersion, processSessionId.